### PR TITLE
Fix documentation for SettingsRow

### PR DESCRIPTION
**Pull request by ianoxley**
Change `description` type to `Node`, and add the missing prop type and description for `className`.

### CODE REVIEW

File: src/components/settings-row/__definition__.js [hunk 1]
```diff
@@ -13,21 +13,23 @@ let definition = new Definition('settings-row', SettingsRow, {
 * Laying out a page in columns? [Try Row](/components/row).
  `,
   propTypes: {
-    children: "Node",
-    description: "String",
-    divider: "Boolean",
-    title: "String"
+    children: 'Node',
+    className: 'String',
+    description: 'Node',
```
Comment (SeanArmstrong) [inline src/components/settings-row/__definition__.js]: Does this still work correctly in the demo site?

File: src/components/settings-row/settings-row.js [hunk 2]
```diff
@@ -60,9 +60,12 @@ class SettingsRow extends React.Component {
      * Heading description
      *
      * @property  description
-     * @type      {Node}
+     * @type      {String|Object}
      */
-    description: PropTypes.node,
+    description: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.object
```
Comment (andrewjtait) [inline src/components/settings-row/settings-row.js]: Just curious about this change? Isn't `node` more flexible?

Comment (ianoxley) [reply src/components/settings-row/settings-row.js]: I suppose so. And I suppose there is a risk of this introducing a breaking change.

However, would the `description` be set to anything other than a string or an JSX object?
And we use `PropTypes.oneOfType([PropTypes.string, PropTypes.object])` elsewhere e.g. in `Heading`: https://github.com/Sage/carbon/blob/master/src/components/heading/heading.js#L36

File: src/components/settings-row/__definition__.js (hunk 1 above)
Comment (SeanArmstrong) [reply src/components/settings-row/__definition__.js]: It does if you add a span to the value

**Pull request merged**